nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3DirectBatchHandler.java

```diff
@@ -143,18 +165,30 @@ public Batch getBatch(String batchId) {
 
         // create the batch
         Batch batch = new Batch(batchId, parameters, getName(), getTransientStore());
-        AssumeRoleRequest assumeRoleRequest = new AssumeRoleRequest().withRoleSessionName(batch.getKey())
-                                                                     .withRoleArn(roleArn);
-        Credentials credentials = stsClient.assumeRole(assumeRoleRequest).getCredentials();
+
+        String bucketPath = Paths.get(bucketPrefix, policyTemplateMatcher.replaceAll(batchId)).toString();
```

Comment:
Don't use `Paths`, this is not a filesystem path.

Reply:
It seems than `policyTemplateMatcher` could be nullable.